src/submission/validation-clinical/surgery.ts

```diff
@@ -80,20 +81,8 @@ export const validate = async (
 			},
 		) as DeepReadonly<Therapy>;
 
-		// if there is surgery with the current submitter_specimen_id submitted before,
-		// or if there are more than one surgeries with the same submitter_specimen_id
-		// in the current submission, then invalid.
 		if (existingSurgeryInDB || surgeries.length > 1) {
-			errors.push(
-				utils.buildSubmissionError(
-					therapyRecord,
-					DataValidationErrors.DUPLICATE_SUBMITTER_SPECIMEN_ID_IN_SURGERY,
-					DonorFieldsEnum.submitter_donor_id,
-					{
-						submitter_specimen_id: therapyRecord[SurgeryFieldsEnum.submitter_specimen_id],
-					},
-				),
-			);
+			checkSurgeryDuplicateOrUpdate(therapyRecord, surgeries, existingSurgeryInDB, errors);
 		} else {
 			validateSurgeryByDonorAndTreatment(therapyRecord, existentDonor, mergedDonor, errors);
```

Comment:
Are these two functions (`checkSurgeryDuplicateOrUpdate` and `validateSurgeryByDonorAndTreatment`) appending items to the `errors` array?

In general we don't want to have functions mutating variables from outside their scope. Better to have these functions return the new items that need to be added to the array and then we can append the new errors. I am writing this comment before reading the internals of your new function, I assume you're simply using the same interface pattern as the existing function - when I see the errors array being passed into this function I would make an assumption that the logic of this method is dependent on the content of this array... so I'll ask naively, does this function care about the existing content of the `errors` array?

I know the `validateSurgeryByDonorAndTreatment` is existing, not added here, but if all of the above is valid and we do want to modify the new function to return errors, it would be best to modify both in the same way (consistent patterns). This way we can improve the code base to be better than when we started.

Reply:
Yeah this is a legacy pattern used across all the validation files, I've been hesitant to refactor b/c it's widespread